src/main.rs

```diff
@@ -221,20 +222,53 @@ fn main() {
     }
 }
 
+struct CommandHistory {}
+impl TypeMapKey for CommandHistory {
+    type Value = HashMap<MessageId, MessageId>;
+}
+
 struct Events {
     cmds: Commands,
 }
 
 impl RawEventHandler for Events {
     fn raw_event(&self, cx: Context, event: Event) {
         match event {
-            Event::Ready(ev) => {
+            Event::Ready(mut ev) => {
                 info!("{} connected to discord", ev.ready.user.name);
+
+                let mut cache = cx.cache.write();
+                cache.update(&mut ev);
+                drop(cache);
+
+                let mut data = cx.data.write();
+                data.insert::<CommandHistory>(HashMap::new());
+                drop(data);
+
                 ban::start_unban_thread(cx);
             }
             Event::MessageCreate(ev) => {
                 self.cmds.execute(cx, &ev.message);
             }
+            Event::MessageUpdate(ev) => {
+                let mut msg = CustomMessage::new();
+
+                msg.id(ev.id)
```

Comment:
I would like to find a way to not respond to every message edit

Reply:
i could check the message timestamps and not respond to messages older than an hour or so

Reply:
also i should probably delete old messages from the history after a certain point

Reply:
You can probably use the ban thread to run the delete code every hour.  https://github.com/rust-lang/discord-mods-bot/blob/master/src/ban.rs#L56

Reply:
To note, we should probably not cache message edit tracking between restarts. It's not valuable enough to UX, and no critical features should be tracking specific message's edits

Reply:
I agree, I was suggesting you use the thread to clear items from the store not the database.